### PR TITLE
Api Samples and Diagrams for 10 more classes / functions

### PR DESCRIPTION
**Pull request by brianegan**
## Description

Adds new assets and code samples to the classes and diagrams listed below:

  - [AlertDialog](https://api.flutter.dev/flutter/material/AlertDialog-class.html)
  - [SimpleDialog](https://api.flutter.dev/flutter/material/SimpleDialog-class.html)
  - [SingleChildScrollView](https://api.flutter.dev/flutter/widgets/SingleChildScrollView-class.html)
  - [showDatePicker](https://api.flutter.dev/flutter/material/showDatePicker.html)
  - [RichText](https://api.flutter.dev/flutter/widgets/RichText-class.html)
  - [InputDecoration](https://api.flutter.dev/flutter/material/InputDecoration-class.html)
  - [Divider](https://api.flutter.dev/flutter/material/Divider-class.html)
  - [StreamBuilder](https://api.flutter.dev/flutter/widgets/StreamBuilder-class.html)
  - [DataTable](https://api.flutter.dev/flutter/material/DataTable-class.html)
  - [BottomSheet](https://api.flutter.dev/flutter/material/BottomSheet-class.html)
    - [showBottomSheet](https://api.flutter.dev/flutter/material/ScaffoldState/showBottomSheet.html)
    - [showModalBottomSheet](https://api.flutter.dev/flutter/material/showModalBottomSheet.html)

## Related Issues

  - https://github.com/flutter/assets-for-api-docs/pull/106

## Tests

No tests necessary.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by brianegan** [comment]
Sorry to ping, @Hixie -- I don't believe these changes require any tests, but I saw the message to contact you in this case and wanted to double-check I wasn't missing any kind of tests that would be necessary for this change since it's just updating documentation with richer examples and diagrams.

File: packages/flutter/lib/src/material/scaffold.dart [hunk 1]
```diff
@@ -1815,6 +1815,47 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
...
+  ///               ),
+  ///             );
+  ///           },
+  ///         );
+  ///       },
+  ///     ),
+  ///   );
+  /// }
+  /// ```
+  ///
```
Comment (Piinks) [inline packages/flutter/lib/src/material/scaffold.dart]:
```suggestion
  /// {@end-tool}
```

File: packages/flutter/lib/src/material/bottom_sheet.dart [hunk 2]
```diff
@@ -441,6 +441,49 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
...
+///               ),
+///             );
+///           },
+///         );
+///       },
+///     ),
+///   );
+/// }
+/// ```
+///
```
Comment (Piinks) [inline packages/flutter/lib/src/material/bottom_sheet.dart]:
```suggestion
/// {@end-tool}
```

File: packages/flutter/lib/src/material/data_table.dart [hunk 3]
```diff
@@ -224,6 +224,67 @@ class DataCell {
 /// target device), it is suggested that you use a
 /// [PaginatedDataTable] which automatically splits the data into
 /// multiple pages.
+///
+/// {@tool snippet --template=stateless_widget_scaffold}
+///
+/// This sample shows how to display a table with three columns: name, age, and
```
Comment (Piinks) [inline packages/flutter/lib/src/material/data_table.dart]:
```suggestion
/// This sample shows how to display a [DataTable] with three columns: name, age, and
```

File: packages/flutter/lib/src/widgets/basic.dart [hunk 4]
```diff
@@ -4994,6 +4994,12 @@ class Flow extends MultiChildRenderObjectWidget {
 ///
 /// {@tool sample}
 ///
+/// This sample demonstrates how to mix and match text with different text
+/// styles using the `RichText` Widget. It displays the text "Hello bold world,"
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/basic.dart]:
```suggestion
/// styles using the [RichText] Widget. It displays the text "Hello bold world,"
```

File: packages/flutter/lib/src/material/divider.dart [hunk 5]
```diff
@@ -22,6 +22,49 @@ import 'theme.dart';
...
+///               child: Text('Below'),
+///             ),
+///           ),
+///         ),
+///       ],
+///     ),
+///   );
+/// }
+/// ```
+///
```
Comment (Piinks) [inline packages/flutter/lib/src/material/divider.dart]:
```suggestion
/// {@end-tool}
```

File: packages/flutter/lib/src/material/dialog.dart [hunk 6]
```diff
@@ -464,6 +466,8 @@ class SimpleDialogOption extends StatelessWidget {
 /// drive a switch statement is that the analyzer will flag any switch statement
 /// that doesn't mention every value in the enum.
 ///
+/// {@animation 350 622 https://flutter.github.io/assets-for-api-docs/assets/material/simple_dialog.mp4}
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dialog.dart]: This will have to be moved above the `{@tool...`

**Review by Piinks — COMMENTED** [review]
Hey @brianegan! Thanks for adding these!

**Comment by brianegan** [comment]
Oh my, how did I miss so many of those `@end-tool`s! Thanks so much. I've committed all of the suggestions and moved the animation out of the `@tool sample`. Please let me know if ya spot anything esle that looks fishy :)

**Review by Piinks — APPROVED** [review]
LGTM! :tada:

**Comment by Hixie** [comment]
> I don't believe these changes require any tests

Sounds like a good test would be a test that catches missing `@end-tool`s :-)

**Comment by brianegan** [comment]
@Hixie Haha, touche!

**Comment by brianegan** [comment]
Good news: The check is already built in! Here's the failure from my first push:

```
src/material/scaffold.dart:1868: Error: Snippet section unterminated.▌11:39:10▐ ELAPSED TIME: 1.181s for bin/cache/dart-sdk/bin/dart /tmp/flutter sdk/dev/bots/analyze-sample-code.dart in .
```

**Pull request merged**